arviz/data/inference_data.py

```diff
@@ -589,6 +586,108 @@ def to_dataframe(
             (dfs,) = dfs.values()
         return dfs
 
+    def to_zarr(self, store=None):
+        """
+        Convert InferenceData to a :class:`zarr.hierarchy.group` using the same groups names
+        as in InferenceData.
+
+        Data groups ("observed_data", "constant_data", "predictions_constant_data") are
+        skipped.
+
+        Raises TypeError if no valid store is found.
+
+        Parameters
+        ----------
+        store: zarr.storage i.e MutableMapping or str, optional
+            Zarr storage class or path to desired DirectoryStore.
+
+        overwrite: bool, optional,
+            If True, delete any pre-existing data in store at path before creating the groups.
+
+        Returns
+        -------
+        zarr.hierarchy.group
+            A zarr hierarchy group containing the InferenceData.
+
+        References
+        ----------
+        https://zarr.readthedocs.io/
+        """
+
+        import zarr
+        from collections.abc import MutableMapping
+
+        # Check store type and create store if necessary
+        if store is None:
+            store = zarr.storage.TempStore(suffix="arviz")
+        elif isinstance(store, str):
+            store = zarr.storage.DirectoryStore(path=store)
+        elif not isinstance(store, MutableMapping):
+            raise TypeError(f"No valid store found: {store}")
+
+        groups = self.groups()
+        if "observed_data" in groups:
+            groups.remove("observed_data")
+        if "constant_data" in groups:
+            groups.remove("constant_data")
+        if "predictions_constant_data" in groups:
+            groups.remove("predictions_constant_data")
```

Comment:
Is this from dataframe code?

There we don't add them to dataframe, but I think zarr can handle these groups.

Reply:
Yes, it is from the `to_dataframe` method. I was not sure what the reasoning behind it was so I just copied it. 
Will remove it in the next commit  :+1: